File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/ProposalService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.ProposalDTO;
import com.hexaware.automobileinsurance.model.Proposal;

public interface ProposalService {

    Proposal createProposal(ProposalDTO dto);

    Proposal getProposalById(Integer proposalId);

    List<Proposal> getAllProposals();

    Proposal updateProposal(Integer proposalId, ProposalDTO dto);

    void deleteProposal(Integer proposalId);
    
    Proposal approveProposal(Integer proposalId);

    Proposal rejectProposal(Integer proposalId);
}

File: automobile-insurance-frontend/src/services/ProposalService.js
import api from "../api/api";

const ProposalService = {

    getAllProposals() {

        return api.get("/proposals");

    },

    getProposalById(id) {

        return api.get(`/proposals/${id}`);

    },

    createProposal(proposal) {

        return api.post("/proposals", proposal);

    },

    updateProposal(id, proposal) {

        return api.put(`/proposals/${id}`, proposal);

    },

    deleteProposal(id) {

        return api.delete(`/proposals/${id}`);

    },

    approveProposal(id){

        return api.put(`/proposals/${id}/approve`);

    },

    rejectProposal(id){

        return api.put(`/proposals/${id}/reject`);

    },
    downloadPolicyDocument(id) {

        return api.get(`/proposals/${id}/policy-document`, {
            responseType: "blob",
        });

    }

};

export default ProposalService;